bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/service/BahmniPatientService.java

```diff
@@ -15,6 +17,8 @@
 
     List<PatientResponse> luceneSearch(PatientSearchParameters searchParameters);
 
+    List<PatientResponse> searchSimilarPatients(PatientSearchParameters searchParameters, PatientResponseMapper patientResponseMapper);
```

Comment:
I think the input parameter to this should actually be PatientProfile (which is https://github.com/openmrs/openmrs-module-emrapi/blob/master/api/src/main/java/org/openmrs/module/emrapi/patient/PatientProfile.java)
The registration create patient actually posts to PatientProfileResource: https://github.com/Bahmni/openmrs-module-bahmniapps/blob/15d0175b14585c7bca47d63f62e61fbf8a412ceb/ui/app/registration/services/defaultPatientServiceStrategy.js#L41

This method should take the same type, so that the UI that is building up a PatientProfile to submit can just submit a partial version.